core/src/avm2/globals/README.md

```diff
@@ -63,6 +63,28 @@ class is loaded.
 
 See `flash/events/Event.as` for an example
 
+## API Versioning
+
+Ruffle supports Flash's Api versioning, which hides newer playerglobal definitions
+(including methods/properties) from SWFs compiled with older API versions.
+For example, see `Event.WORKER_STATE`
+
+To add versioning to an API:
+
+1. Determine the first version where it was added. This can be seen in the Flash Documentation (e.g. "Runtime Versions: Flash Player 11.4, AIR 3.4")
+2. Convert the Flash Player version to an SWF version number using [this chart](http://www.adobe.com/devnet/articles/flashplayer-air-feature-list.html)
+2. Determine the corresponding asc.jar version code for the SWF version. This can be found in avmplus in https://github.com/adobe/avmplus/blob/master/core/api-versions.as
+3. Add an `[API("VersionCode")]` metadata to the defintion. In the `Event.WORKER_STATE` example,
+   this looks like:
+
+   ```actionscript
+   [API("682")]
```

Comment:
not in this PR, but: do you think there's any point in making this less magical, via some preprocessing step maybe?

Reply:
Maybe - I don't know if we'll actually need to implement enough versioned APIs for it to be worth it.